src/library/scala/collection/Map.scala

```diff
@@ -278,7 +278,10 @@ trait MapOps[K, +V, +CC[_, _] <: IterableOps[_, AnyConstr, _], +C]
     *  @return       a new $coll which contains all elements
     *                of this $coll followed by all elements of `suffix`.
     */
-  def concat[V2 >: V](suffix: collection.Iterable[(K, V2)]): CC[K, V2] = mapFactory.from(new View.Concat(toIterable, suffix))
+  def concat[V2 >: V](suffix: collection.IterableOnce[(K, V2)]): CC[K, V2] = mapFactory.from(suffix match {
+    case it: Iterable[(K, V2)] => new View.Concat(toIterable, it)
+    case _ => iterator.concat(suffix.iterator)
```

Comment:
Why not always use `iterator.concat(suffix.iterator)`? What’s the advantage of keeping the `case it: Iterable[(K, V2)]` branch?

Reply:
Using Views is much more efficient for collection types that benefit from multiple traversals when building, so we don't want to drop down to single-traversal Iterators unnecessarily. I can't think of an example where you would want to do this for a Map but all basic methods are implemented in terms of Views, not Iterators, so I'd prefer not to deviate from this scheme unnecessarily.